src/method.jl

```diff
@@ -168,8 +184,14 @@ end
       "percival(::Val{:equ}, nlp) should only be called for equality-constrained problems with bounded variables. Use percival(nlp)",
     )
   end
-  solver = PercivalSolver(nlp, subproblem_modifier = subproblem_modifier)
-  SolverCore.solve!(solver, nlp; subproblem_modifier = subproblem_modifier, kwargs...)
+  subsolver_keys = intersect(keys(subsolver_kwargs), percival_keys)
+  solver_kwargs = Dict(k => subsolver_kwargs[k] for k in subsolver_keys)
+  solver = PercivalSolver(nlp; subproblem_modifier = subproblem_modifier, solver_kwargs...)
+  sub_kwargs = copy(subsolver_kwargs)
+  for k in subsolver_keys
+    pop!(sub_kwargs, k)
+  end
+  SolverCore.solve!(solver, nlp; subproblem_modifier = subproblem_modifier, subsolver_kwargs = sub_kwargs, kwargs...)
```

Comment:
I think you can leave almost everything as it was, except pass the `subsolver_kwargs` to `PercivalSolver` (`solve!` won't need it, right?).
That also mean removing the `percival_keys` definition.

Reply:
Actually, some keywords go to the `PercivalSolver` and some in the `solve!`.
For instance, `:max_cgiter` goes to the `solve!` while `max_radius` goes to `PercivalSolver` (and then to `TronSolver`).
So, passing `max_cgiter` to `PercivalSolver` would return an error.

Reply:
Ah, ok. I misread the `tron` changes. I would have preferred to have something similar in `tron`, but I guess it should be changed when we move to AbstractTrustRegion.
Can you change the name of `percival_keys` to `subsolver_keys` or `trustregion_keys`?

Reply:
Agreed, for updating the changes when moving to `AbstractTrustRegion`.

Reply:
But, there will always be a similar problem as some parameters are for defining the `TronSolver` structure, while others will be for the `solve!` in `tron`.

Reply:
I renamed `percival_keys` as `trustregion_keys`